### PR TITLE
Additional Create/Update Testing

### LINKED ISSUE #245: Adding an Application to the AppGroup after first reconciliation fails
**Describe the bug**
Adding an Application to the AppGroup after first reconciliation fails because there is a call to access the ApplicationStatus of the Application that has not yet gone through reconciliation

```go
func (r *ApplicationGroupReconciler) reconcileApplications(l logr.Logger, appGroup *orkestrav1alpha1.ApplicationGroup) error {
...
        if appGroup.Status.Applications[i].Subcharts == nil {
	        appGroup.Status.Applications[i].Subcharts = make(map[string]orkestrav1alpha1.ChartStatus)
        }
...
}
```

**To Reproduce**
1. Create an ApplicationGroup (i.e. the Bookinfo example)
2. Let the ApplicationGroup finish reconcilation
3. Add an Application to the ApplicationGroup

**Expected behavior**
We should not attempt to init the Application status on first reconciliation and then reference it throughout the reconciliation process. In general, this is bad practice. Additionally, we should be essentially rebuilding this status every time so there is no reason to initialize it only the first reconciliation

### PR DESCRIPTION
**Pull request by jonathan-innis**
- Cleans up some AppGroup reconciliation code
- Resolves a bug with updating applications
- Adds a couple test cases for failed AppGroup and updating the Applications in an AppGroup

Closes #245

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -55,6 +55,7 @@ func main() {
 	var tempChartStoreTargetDir string
 	var disableRemediation bool
 	var cleanupDownloadedCharts bool
+	var runDebug bool
```
Comment (nitishm) [inline main.go:58]: Where does this get used? Couldn't find a reference across the changeset

Comment (jonathan-innis) [reply main.go:58]: 😆 That was from another change that I was moving to a separate PR. I'll remove this

**Comment by jonathan-innis** [comment]
Testing workflow reversal, looks good from my side 👍

**Review by nitishm — APPROVED** [review]
🚀

**Pull request merged**